### ISSUE TITLE
`current-media()` does not interpolate variables, `selector()` does

### ISSUE THREAD

**Issue by cspotcode** [stylus/stylus#2128]
I'd like to fix this bug and am hoping a maintainer can point me in the right direction.

[`selector()`](http://stylus-lang.com/docs/bifs.html#selector) uses [`utils.compileSelectors()`](https://github.com/stylus/stylus/blob/dev/lib/utils.js#L450) to render a full selector string.  It combines nested selectors and performs variable interpolation.  However, [`current-media()`](http://stylus-lang.com/docs/bifs.html#current-media) doesn't.  Based on the way the latter is used in [`+cache()`](https://github.com/stylus/stylus/blob/dev/lib/functions/index.styl#L274), I'm guessing it is supposed to behave like the former?

What's the best way to fix this?  Should I implement a `utils.compileMedia()`?  https://github.com/stylus/stylus/commit/ac404429be0a09668e8b9c56aaea136920086985 seems to be a big media-query refactoring, and a good starting point for my implementation.  Also, should `current-media()` be fixed, or should I add a new BIF that uses `utils.compileMedia()`, leaving `current-media()` intact?

I greatly appreciate any advice you can offer.

```
$foo = 'screen'
@media $foo
    mq: current-media() // expecting '@media screen' but get '@media ($foo)`
```

**Comment by Panya** [comment]
Hi. `compileSelectors` method doesn't interpolate the selectors (they're interpolated in the evaluator, see https://github.com/stylus/stylus/blob/dev/lib/visitor/evaluator.js#L244 and https://github.com/stylus/stylus/blob/dev/lib/visitor/evaluator.js#L1403-L1439).
This method generates nested selectors and resolves relative references (`&`, '/', `^[0]` etc.).

To fix the problem with `current-media` you need to modify its source https://github.com/stylus/stylus/blob/dev/lib/functions/current-media.js. Basically you need to evaluate value of the media query by invoking `this.visit` method (`this` inside every BIF is the reference to current `Evaluator` instance). Something like:

```js
module.exports = function currentMedia(){
  var self = this;
  return new nodes.String(lookForMedia(this.closestBlock.node) || '');

  function lookForMedia(node){
    if ('media' == node.nodeName) {
      node.val = self.visit(node.val); // magic happens here, the `val` property is the unresolved value of the media query (`$foo` ident in an expression in our case).
      return node.toString();
    } else if (node.block.parent.node) {
      return lookForMedia(node.block.parent.node);
    }
  }
};
```

**Comment by Panya** [comment]
Fixed in the `rc-0.54` branch.
